distributedsocialnetwork/author/retrieval.py

```diff
@@ -199,3 +200,126 @@ def get_detailed_author(author_id):
     else:
         # We have this user already, it belongs to our server
         return get_object_or_404(Author, id=author_id)
+
+# Get the github activity of the given author.
+# Creates private posts for every new github
+# event, as authors should only be able to
+# see their own github activity stream.
+def get_github_activity(author_id):
+    author = get_object_or_404(Author, id=author_id)
+    # If the author has set a github profile URL
+    if author.github:
+        githubUsername = author.github.rsplit('/', 1)[-1]
```

Comment:
This assumes the author's github profile url is valid, ie it ends with their github username